DeepHoldout : Fix affects()

danieldresser-ie opened the pull request:
Fix for a little bug @masterkeech found in the recent deep additions.

johnhaddon commented on src/GafferImage/DeepHoldout.cpp:
```diff
@@ -142,10 +142,15 @@ void DeepHoldout::affects( const Gaffer::Plug *input, AffectedPlugsContainer &ou
 	}
 
-	if( input == flattenedPlug()->channelDataPlug() || input == inPlug()->channelNamesPlug() )
+	if( input == flattenedPlug()->channelDataPlug() )
```
We need to keep this line as it was, because `computeChannelData( parent = outPlug() )` does pull on `inPlug()->channelNamesPlug()`.

johnhaddon commented on python/GafferImageTest/DeepHoldoutTest.py:
```diff
@@ -143,5 +143,48 @@ def testBasics( self ):
 		offset["offset"].setValue( imath.V2i( 103, -27 ) )
 		self.assertImagesEqual( holdoutNoZ["out"], cropCleanup["out"], maxDifference = 1e-5 )
 
+	def testDirtyPropogation( self ):
```
Propogation -> Propagation

danieldresser-ie commented:
OK, I think I've addressed all feedback.  Was there something I should change about the commit message format other than adding the "Fixes" tag?  There's nothing for the Changelog here, because it's a fix for a bug that hasn't been released, so there's no user visibility.

danieldresser-ie commented:
Well, bleh.  Now we've got one test failure on Mac OS, in code completely unrelated to this ( though maybe somehow related to the deep image stuff ).  I assume it will go through on a retry, but I figured I should leave it so that you're aware of what is probably a non-deterministic failure in current master.

johnhaddon commented:
> OK, I think I've addressed all feedback.

Thanks!

> Was there something I should change about the commit message format other than adding the "Fixes" tag? There's nothing for the Changelog here, because it's a fix for a bug that hasn't been released, so there's no user visibility.

Nope, that's perfect.

> Well, bleh. Now we've got one test failure on Mac OS, in code completely unrelated to this 

I think I've seen this before, and I think it's a logic error in the test itself. I've restarted the tests here and will merge if they go through, then I'll put up a separate PR to fix the test.
